addons/html_editor/static/tests/paste.test.js

```diff
@@ -3616,29 +3616,44 @@ ${"        "}
 });
 
 describe("onDrop", () => {
-    test("should add text from htmlTransferItem", async () => {
+    test("should drop text from htmlTransferItem before the selected text", async () => {
+        const { el } = await setupEditor("<p>a[b]cd</p>");
+        const pElement = el.firstChild;
+
+        patchWithCleanup(document, {
+            caretPositionFromPoint: () => ({ offsetNode: pElement, offset: 2 }),
+        });
+
+        const dropData = new DataTransfer();
+        dropData.setData("text/html", "b");
+        await dispatch(pElement, "drop", { dataTransfer: dropData });
+        await tick();
+
+        expect(getContent(el)).toBe("<p>acdb[]</p>");
+    });
+    test("should drop text from htmlTransferItem after the selected text", async () => {
         const { el } = await setupEditor("<p>a[b]cd</p>");
         const pElement = el.firstChild;
         const textNode = pElement.firstChild;
 
         patchWithCleanup(document, {
-            caretPositionFromPoint: () => ({ offsetNode: textNode, offset: 3 }),
+            caretPositionFromPoint: () => ({ offsetNode: textNode, offset: 0 }),
         });
 
         const dropData = new DataTransfer();
         dropData.setData("text/html", "b");
         await dispatch(pElement, "drop", { dataTransfer: dropData });
         await tick();
 
-        expect(getContent(el)).toBe("<p>abcb[]d</p>");
+        expect(getContent(el)).toBe("<p>b[]acd</p>");
     });
     test("should not be able to paste inside some branded node", async () => {
         const { el } = await setupEditor(`<p data-oe-model="foo" data-oe-type="text">a[b]cd</p>`);
         const pElement = el.firstChild;
         const textNode = pElement.firstChild;
 
         patchWithCleanup(document, {
-            caretPositionFromPoint: () => ({ offsetNode: textNode, offset: 3 }),
+            caretPositionFromPoint: () => ({ offsetNode: textNode, offset: 0 }),
```

Comment:
Does this need to be changed? The code you change won't even be executed for this case.

Reply:
Done!